### ISSUE TITLE
Test failing after merging pull request #206

### ISSUE THREAD

**Issue by mmclark** [SEED-platform/seed#207]
Here's the stack trace:

```
======================================================================
FAIL: When we have a json encoded list as a column mapping, we concat.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mmclark/git/seed/seed/tests/test_tasks.py", line 275, in test_mapping_w_concat
    mapped_bs.address_line_1, u'1600 Pennsylvania Ave. Someplace Nice'
AssertionError: u'Someplace Nice 1600 Pennsylvania Ave.' != u'1600 Pennsylvania Ave. Someplace Nice'
- Someplace Nice 1600 Pennsylvania Ave.
+ 1600 Pennsylvania Ave. Someplace Nice
```

**Comment by RDmitchell** [comment]
What caused this?

**Comment by StephenCzarnecki** [comment]
Looking into it.

Re running tests:  How do you just run one test?  In this case I thought it
would be like

manage.py test seed.tests.test_tasks:TestTasks.test_mapping_w_concat

but if I run just that the test passes.  However if I run manage.py test
then I get the one failure:

(seed)vagrant@vagrant-ubuntu-trusty-64:~/seed$ ./manage.py test
nosetests --exclude-dir=data_importer --exclude-dir=seed/common
--verbosity=1
Creating test database for alias 'default'...
...........................................................................................................................................................................F...................................................S........SS...........
======================================================================
FAIL: When we have a json encoded list as a column mapping, we concat.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/seed/seed/tests/test_tasks.py", line 275, in
test_mapping_w_concat
    mapped_bs.address_line_1, u'1600 Pennsylvania Ave. Someplace Nice'
AssertionError: u'Someplace Nice 1600 Pennsylvania Ave.' != u'1600
Pennsylvania Ave. Someplace Nice'
- Someplace Nice 1600 Pennsylvania Ave.
+ 1600 Pennsylvania Ave. Someplace Nice


----------------------------------------------------------------------
Ran 245 tests in 91.409s

FAILED (SKIP=3, failures=1)
Destroying test database for alias 'default'...
(seed)vagrant@vagrant-ubuntu-trusty-64:~/seed$ ./manage.py test
seed.tests.test_tasks:TestTasks.test_mapping_w_concat
nosetests seed.tests.test_tasks:TestTasks.test_mapping_w_concat
--exclude-dir=data_importer --exclude-dir=seed/common --verbosity=1
Creating test database for alias 'default'...
.
----------------------------------------------------------------------
Ran 1 test in 5.247s

OK
Destroying test database for alias 'default'...

**Comment by cycustodio** [comment]
How I run individual tests: 
manage.py test <path-to-test-file>

**Comment by StephenCzarnecki** [comment]
That runs all 245 tests.  I am trying to run just the one failing test.  If
I run all the the tests under my debugger A: it takes a while to get to the
failing test and B: I get a lot of exceptions in the passing tests that
popup a browser window.

**Comment by StephenCzarnecki** [comment]
I can run just the tests in the TestTasks suite and get a failing test like
so:

(seed)vagrant@vagrant-ubuntu-trusty-64:~/seed$ ./manage.py test
seed.tests.test_tasks:TestTasks
nosetests seed.tests.test_tasks:TestTasks --exclude-dir=data_importer
--exclude-dir=seed/common --verbosity=1
Creating test database for alias 'default'...
....F.......
======================================================================
FAIL: When we have a json encoded list as a column mapping, we concat.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/seed/seed/tests/test_tasks.py", line 275, in
test_mapping_w_concat
    mapped_bs.address_line_1, u'1600 Pennsylvania Ave. Someplace Nice'
AssertionError: u'Someplace Nice 1600 Pennsylvania Ave.' != u'1600
Pennsylvania Ave. Someplace Nice'
- Someplace Nice 1600 Pennsylvania Ave.
+ 1600 Pennsylvania Ave. Someplace Nice


----------------------------------------------------------------------
Ran 12 tests in 15.305s

FAILED (failures=1)
Destroying test database for alias 'default'...


But if I try to specify the actual test it passes.  Either I am doing
something wrong or there is some dependencies between the tests.  At least
I can cut 245 down to 12 which makes the wait time better.

**Comment by mmclark** [comment]
Here are the django docs on running tests from the command line:

https://docs.djangoproject.com/en/1.7/topics/testing/overview/#running-tests

Essentially it's

./manage.py test app.tests.MyTestCase.test_something

The nose package is also used in SEED, which provides some extensions to the normal django testing framework:

https://nose.readthedocs.org/en/latest/finding_tests.html

**Comment by cycustodio** [comment]
This test might be for concatenating fields (which we are not implementing), also written to pass if fuzzy matching is enabled. 

If you run the parser for each of these you get the following: 

>>> parser.parse("Someplace Nice 1600 Pennsylvania Ave.")
{'house': None, 'street_name': 'Someplace Nice 1600 Pennsylvania', 'suite_num': None, 'street_full': 'Someplace Nice 1600 Pennsylvania Ave', 'street_type': 'Ave', 'suite_type': None, 'other': None}

>>> parser.parse("1600 Pennsylvania Ave. Someplace Nice")
{'house': '1600', 'street_name': 'Pennsylvania', 'suite_num': None, 'street_full': 'Pennsylvania Ave', 'street_type': 'Ave', 'suite_type': None, 'other': 'Someplace Nice'}

What the current code does is combine the following "house" + "street_name" + "street_type". 

We might be ok with commenting out this test or have it fail with a note about which features were enabled for this test to pass.

**Comment by StephenCzarnecki** [comment]
If I just run the single test or just the TestTasks tests then this test
passes with the address in the correct order:  1600 Pennsylvania Ave.
Someplace Nice

If I run all the tests than it fails.  Seems like this indicates there is
some interaction between the tests?

I also tried going back to before I added the new migration and instead of
creating a new migration I re-ran the 0030 migration that updates the BEDES
fields with the new fields and that still fails.

Furthermore sometimes it seems like I can run all of the tests and they all
pass and then I can re-run them again and this test fails.  So far I cannot
reliably reproduce this though.
